scripts/umbrel-os/external-storage/monitor

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# Load monitor-lib
+check_dependencies () {
+  for cmd in "$@"; do
+    if ! command -v $cmd >/dev/null 2>&1; then
+      echo "This script requires \"${cmd}\" to be installed"
+      exit 1
+    fi
+  done
+}
+
+check_dependencies readlink dirname
+UMBREL_ROOT="$(dirname $(readlink -f "${BASH_SOURCE[0]}"))/../../.."
+
+source "${UMBREL_ROOT}/scripts/monitor-lib"
+
+
+# Monitor external mount
+block_device="${1}"
+MOUNT_POINT="${2}"
+
+check_mount () {
+    # Check if Umbrel root is still bind mounted
+    if ! df -h "${UMBREL_ROOT}" | grep --quiet '/dev/sd'; then
+        echo "External device not bind mounted to Umbrel root"
+        return 1
+    fi
+
+    # Check if device is still mounted
+    if ! grep -q "${MOUNT_POINT}" /proc/mounts; then
+        echo "External device not mounted at ${MOUNT_POINT}"
+        return 1
+    fi
+
+    # Check if device is still registered
+    if ! cat /sys/block/${block_device}/ro; then
+        echo "External block device '${block_device}' not found"
+        return 1
+    fi
+
+    # Check if device is read-only
+    READ_ONLY=$(cat /sys/block/${block_device}/ro)
+    if [[ "${READ_ONLY}" == "1" ]]; then
+        echo "External block device '${block_device}' set to 'read-only'"
+        return 1
+    fi
+
+    return 0
+}
+
+run_monitor () {
+    while check_mount; do
+        sleep 1
+    done
+
+    echo "Warning: external drive mount change detected, stopping Umbrel..."
+    systemctl stop umbrel-external-storage
```

Comment:
No need to stop `umbrel-external-storage` here. It's a `Type=oneshot` service, so the script gets executed once on startup, it's not a long running process that needs to be shut down.

Reply:
Ya I noticed that initially, but then some of the processes kicked off by that service continued to run after everything else was brought down which was blocking the un-mounting of the bind mount to get access for stopping everything else ([see comment](https://github.com/getumbrel/umbrel/pull/224#issuecomment-689346819)).

I suspect once we can properly have the monitor exit when the check fails though then this stops being an issue and can be removed (I initially added it as part of my troubleshooting)